src/components/FeatureSetsPanel/FeatureSetsPanelTargetStore/FeatureSetsPanelTargetStore.js

```diff
@@ -7,39 +7,108 @@ import FeatureSetsPanelTargetStoreView from './FeatureSetsPanelTargetStoreView'
-  OTHER_KIND_DEFAULT_FILE_TYPE
+  EXTERNAL_OFFLINE_KIND_DEFAULT_FILE_TYPE,
+  selectedPartitionKindInitialState,
+  isShowAdvancedInitialState,
+  partitionRadioButtonsInitialState,
+  selectedTargetKindInitialState,
+  PARQUET,
+  EXTERNAL_OFFLINE
 } from './featureSetsPanelTargetStore.util'
 
 const FeatureSetsPanelTargetStore = ({
   featureStore,
-  isExternalOfflineTargetsPathValid,
-  setExternalOfflineTargetsPathValid,
+  isTargetsPathValid,
+  setTargetsPathValid,
   setNewFeatureSetTarget
 }) => {
-  const [selectedTargetKind, setSelectedTargetKind] = useState([
-    'offline',
-    'online'
-  ])
-  const [externalOfflineKindData, setExternalOfflineKindData] = useState(
-    externalOfflineKindDataInitialState
+  const [selectedTargetKind, setSelectedTargetKind] = useState(
+    selectedTargetKindInitialState
   )
-  const [offlineKindData, setOfflineKindData] = useState(
-    offlineKindDataInitialState
+  const [selectedPartitionKind, setSelectedPartitionKind] = useState(
+    selectedPartitionKindInitialState
   )
-  const [onlineKindPath, setOnlineKindPath] = useState(
-    onlineKindDataInitialState
+  const [isShowAdvanced, setShowAdvanced] = useState(isShowAdvancedInitialState)
+  const [partitionRadioButtonsState, setPartitionRadioButtonsState] = useState(
+    partitionRadioButtonsInitialState
   )
+  const [data, setData] = useState({
+    online: {
+      name: 'nosql',
+      kind: 'nosql',
+      online: true,
+      path:
+        'v3io:///projects/my-proj/FeatureStore/my-fs/nosql/sets/my-fs-my-tag'
+    },
+    parquet: {
+      name: 'parquet',
+      kind: 'parquet',
+      path:
+        'v3io:///projects/my-proj/FeatureStore/my-fs/parquet/sets/my-fs-my-tag'
+    },
+    externalOffline: { name: 'externalOffline', kind: 'csv', path: '' }
+  })
+
+  const handleAdvancedFieldsLink = kind => {
+    if (
+      !isShowAdvanced[kind] &&
+      selectedPartitionKind[kind].includes('byTime')
+    ) {
+      setNewFeatureSetTarget(
+        featureStore.newFeatureSet.spec.targets.map(targetKind => {
+          if (targetKind.name === kind) {
+            targetKind.time_partitioning_granularity = 'hour'
+          }
 
-  const handleKeyBucketingNumberChange = (key_bucketing_number, kind) => {
-    if (kind === 'externalOffline') {
-      setExternalOfflineKindData(state => ({ ...state, key_bucketing_number }))
+          return targetKind
+        })
+      )
     } else {
-      setOfflineKindData(state => ({ ...state, key_bucketing_number }))
+      setNewFeatureSetTarget(
+        featureStore.newFeatureSet.spec.targets.map(targetKind => {
+          if (targetKind.name === kind) {
+            delete targetKind.time_partitioning_granularity
+            delete targetKind.key_bucketing_number
+            delete targetKind.partition_cols
+          }
+
+          return targetKind
+        })
+      )
+      setData(state => ({
+        ...state,
+        [kind]: {
+          ...checkboxModels[kind].data,
```

Comment:
why do you reset path again if you do this in a first line?
```suggestion
          ...checkboxModels[kind].data,
```